components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -350,21 +350,10 @@ pub fn conv(
     from_base: &Option<Int>,
     to_base: &Option<Int>,
 ) -> Result<Option<Bytes>> {
-    use crate::expr_util::conv::*;
+    use crate::expr_util::conv::conv_internal;
     if let (Some(n), Some(from_base), Some(to_base)) = (n, from_base, to_base) {
-        let from_base = IntWithSign::from_int(*from_base);
-        let to_base = IntWithSign::from_int(*to_base);
-        if is_valid_base(from_base) && is_valid_base(to_base) {
-            let s = String::from_utf8_lossy(n);
-            let s = s.trim();
-            Ok(
-                extract_num_from_str(s, from_base).map_or(Some(b"0".to_vec()), |num| {
-                    Some(num.format_to_base(to_base).into_bytes())
-                }),
-            )
-        } else {
-            Ok(None)
-        }
+        let s = String::from_utf8_lossy(n);
+        Ok(conv_internal(&s, *from_base, *to_base).map(|x| x.into_bytes()))
```

Comment:
How about call it conv directly?

Also how about the non-vectorized version? I think you can simply change it to use the new conv implementation with ease.

Reply:
I' m wondering to change non-vectorized version in this PR or another one. What's your opinion?

Reply:
I think you can do it once, since it's only as simple as a function call I guess.

Reply:
I change `conv_internal` to `conv` but still use an import alias to identify the inner and outer function `conv`.